src/NHibernate.Test/Async/NHSpecificTest/GH1547/Fixture.cs

```diff
@@ -0,0 +1,193 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using NHibernate.Cfg;
+using NHibernate.Driver;
+using NHibernate.Engine;
+using NHibernate.Engine.Query;
+using NHibernate.SqlCommand;
+using NHibernate.SqlTypes;
+using NHibernate.Util;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH1547
+{
+	using System.Threading.Tasks;
+	using System.Threading;
+	[TestFixture, Explicit("Contains only performances benchmark")]
+	public class FixtureAsync : BugTestCase
+	{
+		protected override void Configure(Configuration configuration)
+		{
+			base.Configure(configuration);
+
+			var driverClass = ReflectHelper.ClassForName(configuration.GetProperty(Cfg.Environment.ConnectionDriver));
+			DriverForSubstitutedCommand.DriverClass = driverClass;
+
+			configuration.SetProperty(
+				Cfg.Environment.ConnectionDriver,
+				typeof(DriverForSubstitutedCommand).AssemblyQualifiedName);
+		}
+
+		protected override void DropSchema()
+		{
+			(Sfi.ConnectionProvider.Driver as DriverForSubstitutedCommand)?.CleanUp();
+			base.DropSchema();
+		}
+
+		[Test]
+		public Task SimpleLinqPerfAsync()
+		{
+			return BenchmarkAsync(
+				"Simple LINQ",
+				s =>
+					s
+						.Query<Entity>()
+						.Where(e => e.Name == "Bob"));
+		}
+
+		[Test]
+		public Task LinqWithNonParameterizedConstantPerfAsync()
+		{
+			return BenchmarkAsync(
+				"Non parameterized constant",
+				s =>
+					s
+						.Query<Entity>()
+						.Where(e => e.Name == "Bob")
+						.Select(e => new { e, c = 2 }));
+		}
+
+		[Test]
+		public Task LinqWithListParameterPerfAsync()
```

Comment:
Try-catch optimization in tests looks ugly. Maybe we need to consider option to disable it for tests.

Reply:
Why not, but I would see that as just a nice to have.